src/Middleware/HttpLogging/src/HttpLoggingFields.cs

```diff
@@ -135,6 +135,12 @@ public enum HttpLoggingFields : long
     /// </summary>
     ResponseBody = 0x800,
 
+    /// <summary>
+    /// Log how long it took to process the request and return a response in total milliseconds.
+    /// </summary>
+    // TODO: This does not include the time taken to write the response body.
+    Duration = 0x1000,
```

Comment:
Would moving writing HTTP logging to `HttpResponse.OnCompleted` help?

Reply:
I think we'd still need the interceptor to run on first write, but then we'd delay the log until the middleware exited.
https://github.com/dotnet/aspnetcore/pull/48679/files#diff-04d04600ca956dde608297e0c6d37c5904df7528e385c6cb5f012b6f5d85a4f5R200